sql/src/test/java/org/apache/druid/sql/calcite/util/SpecificSegmentsQuerySegmentWalker.java

```diff
@@ -78,9 +78,7 @@ public SpecificSegmentsQuerySegmentWalker add(
   )
   {
     final Segment segment = new QueryableIndexSegment(index, descriptor.getId());
-    if (!timelines.containsKey(descriptor.getDataSource())) {
-      timelines.put(descriptor.getDataSource(), new VersionedIntervalTimeline<>(Ordering.natural()));
-    }
+    timelines.putIfAbsent(descriptor.getDataSource(), new VersionedIntervalTimeline<>(Ordering.natural()));
```

Comment:
Another good candidate for `computeIfAbsent`.

Reply:
Done.